### DISCUSSION TITLE
Usage of Build_Profile_Kind for compiler switches

### DISCUSSION THREAD

**Discussion by captain-haddock17** [alire-project/alire#1055] [Q&A]
Hello Dear Team,

I'd like to use `Build_Profile_Kind` to define alternative compiler switches.

As this variable is defined in the (generated) `*_config.gpr` file, my specific setup could be erased  by 'alr' at some point.

My usage is 

```Ada
   Ada_Compiler_Default_Switches := ...

   Ada_Compiler_Release_Switches := Ada_Compiler_Default_Switches & (
           "-O3" );

   Ada_Compiler_Debug_Switches := Ada_Compiler_Default_Switches & (
            "-Og" -- Optimize for debug
           ,"-g" -- Generate debug info
   );
   Ada_Compiler_Profile_Switches := Ada_Compiler_Default_Switches & (
            "-Og" -- Optimize for debug
           ,"-g" -- Generate debug info
           ,"-fstack-usage" -- generate an extra file that specifies the maximum amount of stack used
   );
   
   case Build_Profile is
         when "Debug"  =>
            Ada_Compiler_Switches := Ada_Compiler_Debug_Switches;
         when "Profile"  => 
            Ada_Compiler_Switches := Ada_Compiler_Profile_Switches;
         when "Release" => 
            Ada_Compiler_Switches := Ada_Compiler_Release_Switches;
         when others => null;
   end case;
```

I did not like to put this in `myProject.gpr` , because it would redundant a lot of config lines.

Would there be an elegant way to do this ?

Thanks a lot,
Kind regards,
William

**Comment by Fabien-Chouteau** [comment]
Hello @captain-haddock17

If you want to tweak the build switches it's better to do it from the `alire.toml` manifest.
For instance:
```toml
[build-switches]
release.optimization = ["-O1", "-gnatn"]
```

I'd like to know what the custom set of switches doesn't work for you.

**Reply by kevlar700** [reply]
I'm not sure from the Alire docs what release.optimization does? It seems to apply to both alr build --development and --release but maybe I have broken the profiles. Is it meant to add switches to all builds?

If so is the way that SPARKNACl does switches a good way to have various debug settings. I assume it has to be done per dependency crate as well.

https://github.com/rod-chapman/SPARKNaCl/blob/master/sparknacl.gpr
